### DISCUSSION TITLE
builtin virtiofsd-rs

### DISCUSSION THREAD

**Discussion by muhamadazmy** [cloud-hypervisor/cloud-hypervisor#3112] [Ideas]
Right now to use `--fs` flag in cloud-hypervisor I actually how to start and maintain a separate process that runs `virtiofsd-rs` with the correct command line arguments, and such.

It gets messy when i need to share multiple directories, and if one of them crashed for any reason.

I am actually wondering, why this is needed, since virtiofsd-rs is another rust crate that can probably get impeded in cloud-hypervisor itself, so I need to only provide the tag and the filesystem path as arguments. 

I am sure this is not an original idea, and you probably discussed it internally, but would be nice to also share why it's done like this and not built-in cloud-hypervisor itself.

**Comment by rbradford** [comment]
The separate process model with vhost-user as a protocol in between is to ensure stability (and to a lesser extent) security. If the virtiofsd daemon crashes then it doesn't take down the VMM.

**Reply by muhamadazmy** [reply]
Is there a way to somehow reconnect from the vm to a new virtiofsd process (same of different unix socket) if the virtiofsd crashes ? or that's the end of the filesystem inside the vm ?

**Reply by sboeuf** [reply]
This is not supported because we would need some special handling between the daemon and the VMM. Cloud Hypervisor supports global disconnection/reconnection of any vhost-user backend but in case of virtio-fs, the backend is not equipped for it.
You should get in touch with virtio-fs maintainers to ask what's the status of supporting a crash, but I don't think there's a solution yet.
